tiatoolbox/wsicore/wsireader.py

```diff
@@ -3832,6 +3718,398 @@ def read_rect(
         pad_mode: str = "constant",
         pad_constant_values: int | IntPair = 0,
         coord_space: str = "baseline",
+        **kwargs: dict,
+    ) -> np.ndarray:
+        """See TIFFWSIReaderDelegate.read_rect docs for details."""
+        return self.tiff_reader_delegate.read_rect(
+            location,
+            size,
+            resolution,
+            units,
+            interpolation,
+            pad_mode,
+            pad_constant_values,
+            coord_space,
+            **kwargs,
+        )
+
+    def read_bounds(
+        self: TIFFWSIReader,
+        bounds: IntBounds,
+        resolution: Resolution = 0,
+        units: Units = "level",
+        interpolation: str = "optimise",
+        pad_mode: str = "constant",
+        pad_constant_values: int | IntPair = 0,
+        coord_space: str = "baseline",
+        **kwargs: dict,
+    ) -> np.ndarray:
+        """See TIFFWSIReaderDelegate.read_bounds docs for details."""
+        return self.tiff_reader_delegate.read_bounds(
+            bounds,
+            resolution,
+            units,
+            interpolation,
+            pad_mode,
+            pad_constant_values,
+            coord_space,
+            **kwargs,
+        )
+
+
+class FsspecJsonWSIReader(WSIReader):
+    """Reader for fsspec zarr json generated by: tiatoolbox/utils/tiff_to_fsspec.py.
+
+    The fsspec zarr json file represents a SVS or TIFF file
+    that be accessed using byte range HTTP API.
+
+    All the information on the chunk locations in the SVS or TIFF file
+    is outlined as byte-ranges in the JSON,
+    so the reader requests only chunks that are needed to display requested tiles,
+    rather than the entire SVS or TIFF file.
+    """
```

Comment:
```suggestion

        """
```

Please can you use [google docstring style](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html) for docstrings?